src/include/wiredtiger.in

```diff
@@ -3682,6 +3704,502 @@ struct __wt_extractor {
+ *
+ */
+struct __wt_file_handle {
+	/*!
+	 * The name of the file.
+	 */
+	char *name;
+
+	/*!
+	 * Close a file handle; the WT_FILE_HANDLE will not be further accessed
+	 * by WiredTiger.
+	 *
+	 * This method is not called concurrently with other file system or
+	 * handle operations.
+	 *
+	 * @errors
+	 *
+	 * @param file_handle the WT_FILE_HANDLE
+	 * @param session the current WiredTiger session
+	 */
+	int (*close)(WT_FILE_HANDLE *file_handle, WT_SESSION *session);
+
+	/*!
+	 * Indicate expected future use of file ranges, based on the POSIX
+	 * 1003.1 standard fadvise.
+	 *
+	 * This method is not required, and should be set to NULL when not
+	 * supported by the file.
+	 *
+	 * This method may be called concurrently with other file system or
+	 * handle operations.
+	 *
+	 * @errors
+	 *
+	 * @param file_handle the WT_FILE_HANDLE
+	 * @param session the current WiredTiger session
+	 * @param offset the file offset
+	 * @param len the size of the advisory
+	 * @param advice one of ::WT_FILE_HANDLE_WILLNEED or
+	 *    ::WT_FILE_HANDLE_DONTNEED.
+	 */
+	int (*fadvise)(WT_FILE_HANDLE *file_handle,
+	    WT_SESSION *session, wt_off_t offset, wt_off_t len, int advice);
+
+	/*!
+	 * Ensure disk space is allocated for the file, based on the POSIX
+	 * 1003.1 standard fallocate.
+	 *
+	 * This method is not required, and should be set to NULL when not
+	 * supported by the file.
+	 *
+	 * This method is not called concurrently with other file system or
+	 * handle operations. If the file handle's fallocate function can
+	 * support fallocate calls concurrently with other file system or
+	 * handle operations, set the WT_FILE_HANDLE::fallocate_nolock method
+	 * instead.
+	 *
+	 * @errors
+	 *
+	 * @param file_handle the WT_FILE_HANDLE
+	 * @param session the current WiredTiger session
+	 * @param offset the file offset
+	 * @param len the size of the advisory
+	 */
+	int (*fallocate)(WT_FILE_HANDLE *file_handle,
+	    WT_SESSION *session, wt_off_t, wt_off_t);
+
+	/*!
+	 * Ensure disk space is allocated for the file, based on the POSIX
+	 * 1003.1 standard fallocate.
+	 *
+	 * This method is not required, and should be set to NULL when not
+	 * supported by the file.
+	 *
+	 * This method may be called concurrently with other file system or
+	 * handle operations. If the file handle's fallocate function cannot
+	 * support fallocate calls concurrently with other file system or
+	 * handle operations, set the WT_FILE_HANDLE::fallocate method instead.
+	 *
+	 * @errors
+	 *
+	 * @param file_handle the WT_FILE_HANDLE
+	 * @param session the current WiredTiger session
+	 * @param offset the file offset
+	 * @param len the size of the advisory
+	 */
+	int (*fallocate_nolock)(WT_FILE_HANDLE *file_handle,
+	    WT_SESSION *session, wt_off_t, wt_off_t);
+
+	/*!
+	 * Lock/unlock a file from the perspective of other processes running
+	 * in the system.
+	 *
+	 * This method is not required, and should be set to NULL when not
+	 * required by the system.
```

Comment:
What does it mean for lock not to work if the database is opened for updates? I expected lock to be required for filesystems that support writes.